frontend/src/plugin/registry.tsx

```diff
@@ -623,4 +624,26 @@ export function registerHeadlampEventCallback(callback: HeadlampEventCallback) {
   store.dispatch(addEventCallback(callback));
 }
 
+/**
+ * Register a plugin settings component.
+ * @param name - The name of the plugin.
+ * @param component - The component to use for the settings.
+ * @param displaySaveButton - Whether to display the save button.
+ *
+ * @example
+ *
+ * ```tsx
+ * import { registerPluginSettings } from '@kinvolk/headlamp-plugin/lib';
+ *
+ * registerPluginSettings('my-plugin', MyPluginSettingsComponent, true);
+ * ```
+ */
+export function registerPluginSettings(
```

Comment:
Please add a section to docs/development/plugins/functionality.md following the format of the other pieces in that document.

See the existing ones there for how it should be written. (In short... It links to the API reference, and also the example plugin with the small description of the functionality.)